crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -1359,13 +1359,10 @@ pub(crate) fn build_transaction_receipt_with_block_receipts(
 
     #[cfg(feature = "optimism")]
     {
-        let mut op_fields = OptimismTransactionReceiptFields {
-            deposit_nonce: receipt.deposit_nonce.map(U64::from),
-            ..Default::default()
-        };
+        let mut op_fields = OptimismTransactionReceiptFields::default();
 
-        if let Some(l1_block_info) = optimism_tx_meta.l1_block_info {
-            if !transaction.is_deposit() {
+        if !transaction.is_deposit() {
```

Comment:
Yes, ofc. I just wasn't sure if the l1_block_info is always present when the transaction is of type deposit.
And I moved op_fields.deposit_nonce and op_fields.deposit_receipt_version because of this #6604